lib/api.js

```diff
@@ -240,31 +241,46 @@ async function handleBranchUpdate(context, eventName, event) {
 }
 
 async function handleScheduleTrigger(context) {
-  const { octokit } = context;
+  const { octokit, config: { projectColumnNumber } } = context; 
+  if (typeof projectColumnNumber === 'undefined') {
+    throw new Error('Project column number is undefined. Skipping scheduled job.');
+  }
 
-  const { GITHUB_REPOSITORY } = process.env;
-  const [owner, repo] = (GITHUB_REPOSITORY || "").split("/", 2);
+  logger.debug(`Listing issues in column ${projectColumnNumber} ...`);
+  const { data: cards } = await octokit.projects.listCards({
+    column_id: projectColumnNumber,
+  });
 
-  if (!owner || !repo) {
-    throw new Error(`invalid GITHUB_REPOSITORY value: ${GITHUB_REPOSITORY}`);
+  if (cards.length === 0) {
+    logger.info(`Column ${projectColumnNumber} doesn't have any issues. Cancelling automerge job.`);
+    return;
   }
 
-  logger.debug("Listing pull requests ...");
-  const { data: pullRequests } = await octokit.pulls.list({
-    owner,
-    repo,
-    state: "open",
-    sort: "updated",
-    direction: "desc",
-    per_page: MAX_PR_COUNT
-  });
+  const pullRequestNumbers = cards.reduce((acc, card) => {
+    const splittedUrl = card.content_url.split('/');
+    const pullRequestRepo = splittedUrl[splittedUrl.length - 3];
```

Comment:
This can potentially fail badly if the url has different format than anticipated. Maybe add check for parts size?